### PR TITLE
add negative path constraints

### PR DESCRIPTION
**Pull request by Rahien**
allow for negative path constraints. This means that the constraint will only match triples if a given path does NOT exist. E.g.
```
ext:formListingS a form:Scope;
  sh:path form:includes;
  form:constraint [
     a sh:NodeShape ;
     sh:property [
       sh:path rdf:type ;
       sh:targetNode form:Listing ;
     ];
  ];
  form:constraint [
	a sh:NodeShape ;
	sh:not [
		a sh:PropertyShape ;
		sh:path ex:property ;
                sh:targetNode form:ListingTable ;
	]; 
  ];
```
Is a scope that will match uris with the rdf:type form:Lising that do NOT also have the type form:ListingTable

### CODE REVIEW

File: src/triples-for.js [hunk 1]
```diff
@@ -220,23 +210,75 @@ export function triplesForScope(scopeUri, options) {
...
+}
+
+function getNegativeNodeShapeConstraints(constraints, scopeOptions) {
+  const { store, formGraph } = scopeOptions;
+  return toPathConstraints(
+    constraints.filter((constraint) =>
+      store.any(
+        constraint.object,
+        RDF("type"),
+        SHACL("NegativeNodeShape"),
```
Comment (cecemel) [inline src/triples-for.js:233]: Is this that thing in SHACL?

Comment (cecemel) [reply src/triples-for.js:233]: I see NotConstraintComponent?

Comment (Rahien) [reply src/triples-for.js:233]: totally not SHACL, seems they want us to use something like this:
```
ex:NotExampleShape
	a sh:NodeShape ;
	sh:targetNode ex:InvalidInstance1 ;
	sh:not [
		a sh:PropertyShape ;
		sh:path ex:property ;
		sh:minCount 1 ;
	] .
```
which is fair, will change to that

**Review by cecemel — APPROVED** [review]
Thanks added value!
thanks for refactoring a bit
And it seems no stuff broke.

Note: it might be useful in the ember submission form fields to make a little test form in the test app to show how to use it.
Note 2: I do think some tests in this library might be of help. Kinda complicated code.

**Pull request merged**